azurerm/resource_arm_private_link_service.go

```diff
@@ -0,0 +1,1006 @@
+						},
+						"enable_ip_forwarding": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"hosted_workloads": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"ip_configurations": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"name": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"location": azure.SchemaLocation(),
+						"mac_address": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"network_security_group": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"location": azure.SchemaLocation(),
+									"name": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"tags": tags.Schema(),
+									"type": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"primary": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"private_endpoint": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"location": azure.SchemaLocation(),
+									"name": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"tags": tags.Schema(),
+									"type": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"resource_guid": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"tags": tags.Schema(),
+						"tap_configurations": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"name": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"type": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"virtual_machine": {
```

Comment:
maybe we can flatten this to just `virtual_machine_id`?

Reply:
Removed unnecessary field